main.cpp

```diff
@@ -51,22 +52,401 @@ If you need inspiration for what to write, take a look at previously approved st
+    }
+}
+void SchoolDay::shortenSchoolDay( float lengthDecrease )
+{
+    if (halfDay == false)
+    {
+        length -= lengthDecrease;
+        std::cout << "SchoolDay length is now: " << length << std::endl;
+        halfDay = true;
+    }
+    
+}   
+void SchoolDay::sunCameOut()
+{
+    if (isRaining == true)
+    {
+        isRaining = false;
+        std::cout << "sun came out" << std::endl;
+    }
+    else
+    {
+        std::cout << "sun is already out" << std::endl;
+    }
+}
 
 /*
  UDT 3:
  */
+struct SurfReport
+{
+    SurfReport();
+    ~SurfReport();
+    float waveHeight;
+    float wavePeriod;
+    std::string windDirection;
+    int windSpeed;
+    bool isRaining = false;
+    bool isBusy = false;
+    struct Surfer
+    {
+        Surfer();
+        ~Surfer();
+        bool atBeach = true;
+        int surferAge;
+        int surferSkill = 0;
+        float boardSize = 6.8f;
+        bool havingLesson = false;
+
+        void levelUp( int improvementAmount );
+        void getTubed( int currentStokeLevel );
+        void wipeOut( int currentStokeLevel );
+    };
+
+    void increaseWaveHeight( float newWaveHeight );
+    bool shouldIGoOut( int fatigueLevel );
+    float chooseBoardSize();
+};
+
+SurfReport::SurfReport()
+{
+    waveHeight = 3.5f;
+    wavePeriod = 18.f;
+    windDirection = "SSE";
+    windSpeed = 10;
+    std::cout<< "a SurfReport was created" << std::endl;
+}
+
+SurfReport::~SurfReport()
+{
+    std::cout<< "a SurfReport was destroyed" << std::endl;
+}
+
+SurfReport::Surfer::Surfer() :
+surferAge(35)
+{
+    std::cout<< "a Surfer aged: " << surferAge << " was created" << std::endl;
+}
+
+SurfReport::Surfer::~Surfer()
+{
+    std::cout<< "a Surfer aged: " << surferAge << " was destroyed" << std::endl;
+}
+
+void SurfReport::Surfer::levelUp( int improvementAmount )
+{
+    if (surferSkill <= 10)
+    {
+        for (int x = 0; x <= improvementAmount && surferSkill <= 10; x++)
+
+            surferSkill += 1; 
+    }
+}
+void SurfReport::Surfer::getTubed( int currentStokeLevel )
+{
+    currentStokeLevel += 10; 
+    std::cout << "tubed! Stoke level = " << currentStokeLevel << std::endl;
+}
+void SurfReport::Surfer::wipeOut( int currentStokeLevel )
+{
+    currentStokeLevel -=10;
+    std::cout << "wipeout! Stoke level = " << currentStokeLevel << std::endl;
+}
+
+void SurfReport::increaseWaveHeight( float heightIncrease )
+{
+    waveHeight += heightIncrease; 
+}
+bool SurfReport::shouldIGoOut( int fatigueLevel )
+{
+    if (fatigueLevel < 10)
+    {
+        fatigueLevel += 1;
+        std::cout << "new fatigue level = " << fatigueLevel << std::endl;
+        std::cout << "yes you should go out" << std::endl;
+        return true;
+    }
+    std::cout << "Don't go out" << std::endl;
+    return false;
+}
+float SurfReport::chooseBoardSize()
+{
+    float boardLength = 8.f;
+    for (float x = 0.f; x <= waveHeight; x+=1.f)
+    {
```

Comment:
add space around math operators, including `+=`
this will make the code more readable
- [ ] revise